### ISSUE TITLE
VerticalOptions=FillAndExpand 

### ISSUE THREAD

**Issue by newdeal30** [robinmanuelthiel/flexbutton#10]
Hi!
I'm using Flexbutton with in a Grid and want to set the height with percentage. 
FlexButton wants to get a HeightRequest - VerticalOptions is not working.
Tried to set HeightRequest to a more bigger size - Than button has the correct size, but text is no align correctly
Would be a nice

Thanks

**Comment by robinmanuelthiel** [comment]
Hi Michael, unfortunately, I can not reproduce your issue. I have played around with the FlexButton in Grids today and for me everything works as expected. Could you provide me a code sample of what you are trying to achieve with screenshots of the current and expected results?

**Comment by Jyosua** [comment]
I encountered this today, I think:

```xml
<Grid>
    <Grid.ColumnDefinitions>
        <ColumnDefinition Width="auto"/>
        <ColumnDefinition Width="*" />
    </Grid.ColumnDefinitions>
    <Grid.RowDefinitions>
        <RowDefinition Height="*" />
        <RowDefinition Height="auto" />
        <RowDefinition Height="auto" />
        <RowDefinition Height="auto" />
        <RowDefinition Height="40" />
        <RowDefinition Height="*" />
    </Grid.RowDefinitions>
    <Label Grid.Row="1" Grid.Column="0" x:Name="NumberLabel" Margin="4,4,0,0" FontSize="14" TextColor="#37404A" />
    <local:HtmlLabel Grid.Row="1" Grid.Column="1" x:Name="QuestionTitleView" FontSize="20" HorizontalOptions="FillAndExpand" TextColor="#37404A" />
    <Label Grid.Row="2" Grid.Column="1" x:Name="SubtitleLabel" FontSize="14" TextColor="#616569" IsVisible="false" />
    <!--<Button Grid.Row="4" Grid.Column="1" x:Name="ButtonView" Text="Next" WidthRequest="100" 
            HorizontalOptions="End" VerticalOptions="Center" Margin="0,0,5,0" 
            BackgroundColor="#37404A" TextColor="#A8B4BB" FontSize="20" 
            IsVisible="{Binding OKButtonIsVisible, Mode=TwoWay}" />-->
    <flex:FlexButton Grid.Row="4" Grid.Column="1" x:Name="ButtonView" Text="Next"
                     WidthRequest="100" HeightRequest="40" 
                     HorizontalOptions="End" VerticalOptions="Center"
                     CornerRadius="5" Margin="0,0,5,0" FontSize="20" 
                     ForegroundColor="#A8B4BB" HighlightForegroundColor="#A8B4BB"
                     BackgroundColor="#2A313A" HighlightBackgroundColor="#4E5866"
                     IsVisible="{Binding OKButtonIsVisible, Mode=TwoWay}" />
</Grid>
```

If I don't set the HeightRequest of the flexbutton, it does not inherit the 40 I set for the grid row like a normal Button does. If I do set the HeightRequest to 40, the flexbutton has the correct height, but the text label is at the bottom of the button, rather than centered.

**Comment by Depechie** [comment]
@Jyosua for the text being on the bottom of the flexbutton instead of centered > try setting the padding of your button to "0,0,0,0".
There is a default value for the padding in the flexbutton, that is ok for larger buttons, but not all to good for smaller ( in height ) buttons.
Maybe something @robinmanuelthiel can look at in the future.

**Comment by Jyosua** [comment]
Thank you @Depechie! You are indeed correct!

**Comment by robinmanuelthiel** [comment]
Maybe a related error, but I can reproduce, that the text is not vertically centered on smaller Buttons!
![image](https://user-images.githubusercontent.com/5874465/37991465-e9fad70c-3211-11e8-80db-a44f578795fc.png)

**Comment by Jyosua** [comment]
That is what was happening to me, yeah. I tried using FillAndExpand within my grid after setting that padding to 0 and it then respects the height setting of the grid row and centers the text. So really, I think it not centering on small buttons due to the default padding being wrong is the only issue here.

**Comment by robinmanuelthiel** [comment]
Should be fixed in latest release 0.5.0
